packages/compass-collection/src/stores/collection-tab.ts

```diff
@@ -86,6 +106,28 @@ export function activatePlugin(
 
   void collectionModel.fetchMetadata({ dataService }).then((metadata) => {
     store.dispatch(collectionMetadataFetched(metadata));
+
+    // Assign experiment for Mock Data Generator:
+    // Only assign when experimentationServices.assignExperiment is initialized,
+    // we're connected to Atlas,
+    // and the org-level setting for AI features is enabled
+    if (
+      experimentationServices?.assignExperiment && // Ensures experimentation services are available
```

Comment:
As you are providing default context value, in what case are they not available?